### PR TITLE
Feat/turn text to timestamp

### PR DESCRIPTION
**Pull request by rksht**
Select text that has form `hh:mm:ss` and turn it into timestamp.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +68,7 @@
         "slate-auto-replace": "^0.12.1",
         "slate-react": "0.21.24",
         "slug": "1.1.0",
-        "typescript": "3.5.2"
+        "typescript": "^3.4.5"
```
Comment (ciju) [inline package.json]: downgrading typescript?

File: src/keycodeMap.ts [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+	mark_selection_as_timestamp: 'Ctrl-t',
+	turn_text_to_timestamp: 'Ctrl-y',
+	put_timestamp: 'Ctrl-Shift-t',
+	seek_to_timestamp: 'Ctrl-g',
+	capture_frame: 'Ctrl-i',
+	debug_print: 'Ctrl-d',
 };
+
+const commandNameOfKeycode = fromPairs(
+	Object.entries(keycodeOfCommandName).map(([commandName, keycode]) => [keycode, commandName])
```
Comment (ciju) [inline src/keycodeMap.ts]: `Object.entries(keycodeOfCommandName).map(([commandName, keycode]) => [keycode, commandName])`
or
`Object.entries(keycodeOfCommandName).map(([cmd, key]) => [key, cmd])` 
?

Try to make code easier to read.

File: src/utils/fromPairs.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export default function fromPairs<TValue>(iterable: Iterable<[string, TValue]>)
```
Comment (ciju) [inline src/utils/fromPairs.ts]: Whats this complicated logic about? The loop issue is because of this function. If my understanding is correct, the idea is to get an object of of key, val pairs. Its a simple array.reduce function away. Why have complicated difficult to parse code?

File: src/EditorComponent.jsx [hunk 4]
```diff
@@ -12,13 +12,13 @@ import { makeStyles, createStyles } from '@material-ui/core/styles';
```
Comment (ciju) [inline src/EditorComponent.jsx]: This file is becoming complex. Lets try to simplify it.

File: src/EditorComponent.jsx [hunk 5]
```diff
@@ -45,127 +45,66 @@ const EditorComponent = props => {
...
-    const tellExtensionToCaptureFrame = (_state, _dispatch) => {
-      const { videoId } = parentApp.currentVideoInfo();
-      if (videoId) {
-        window.frames[0].postMessage({ type: AppConfig.CaptureCurrentFrameMessage }, '*');
-      }
-      return true;
-    };
+    // Map from keycode to command function
+    const keymapObject = Object.fromEntries(
+      Object.entries(commandNameOfKeycode).map(([keycode, cmdName]) => {
```
Comment (ciju) [inline src/EditorComponent.jsx]: Use shorter names.

File: package.json [hunk 6]
```diff
@@ -1,83 +1,88 @@
...
+	"version": "0.1.0",
+	"private": true,
+	"dependencies": {
+		"@emotion/core": "10.0.10",
+		"@emotion/styled": "10.0.10",
+		"@kenshooui/react-multi-select": "^1.1.3",
+		"@material-ui/core": "4.1.1",
+		"@material-ui/icons": "4.2.0",
+		"@material-ui/lab": "^4.0.0-alpha.16",
+		"@slate-editor/font-size-plugin": "^5.1.0",
```
Comment (ciju) [inline package.json]: Why is this included?

File: package.json [hunk 7]
```diff
@@ -1,83 +1,88 @@
...
-        "not ie <= 11",
-        "not op_mini all"
-    ]
+	"name": "vid-annot-slate",
+	"version": "0.1.0",
+	"private": true,
+	"dependencies": {
+		"@emotion/core": "10.0.10",
+		"@emotion/styled": "10.0.10",
+		"@kenshooui/react-multi-select": "^1.1.3",
```
Comment (ciju) [inline package.json]: Why is this included?

Comment (rksht) [reply package.json]: Was trying out something. Removing.

File: package.json (hunk 6 above)
Comment (rksht) [reply package.json]: Could be due to the rebase. Removing.

File: src/EditorComponent.jsx [hunk 8]
```diff
@@ -53,119 +53,52 @@ const EditorComponent = props => {
...
+        return [keycode, cmdFunctions[cmdName]];
+      })
+    );
+
+    const autosavePlugin = new Plugin({
+      view: editorView => new AutosavePlugin(editorView, videoId),
+    });
+
+    const content = document.getElementById('__content__');
```
Comment (ciju) [inline src/EditorComponent.jsx]: Whats this for?

File: src/keycodeMap.ts (hunk 2 above)
Comment (ciju) [reply src/keycodeMap.ts]: Making this change?

File: src/EditorComponent.jsx (hunk 8 above)
Comment (rksht) [reply src/EditorComponent.jsx]: Removing. It was there for test content.
